src/test/button-creation.test.ts

```diff
@@ -0,0 +1,93 @@
+const {ButtonCreation} = require('../form-element-creator/button-creation')
+
+test('create submit button ', () => {
+    const propertiesConfig = {
+        buttons: [
+            {
+                class: 'buttons',
+                name: 'submit',
+                type: 'submit',
+                value: 'Submit'
+            }
+        ]
+    };
+
+    const mockClickHandler = jest.fn();
+    const buttonCreation = new ButtonCreation();
+    const button = buttonCreation.create(propertiesConfig.buttons[0].value, mockClickHandler);
+
+    button.click();
+
+    expect(button.textContent).toEqual('Submit');
+    expect(mockClickHandler).toHaveBeenCalled();
+
+});
+
+
+test('create previous button ', () => {
+    const propertiesConfig = {
+        buttons: [
+            {
+                class: 'buttons',
+                name: 'previous',
+                type: 'previous',
+                value: 'Previous'
+            }
+        ]
+    };
+
+    const mockClickHandler = jest.fn();
+    const buttonCreation = new ButtonCreation();
+    const button = buttonCreation.create(propertiesConfig.buttons[0].value, mockClickHandler);
+
+    button.click();
+
+    expect(button.textContent).toEqual('Previous');
```

Comment:
expect(button.class).toEqual('buttons');
.......
......

Reply:
oprocz given when then to dorzuc jeszcze to o czym rozmawialismy i jest wyzej w komentarzu